### ISSUE TITLE
Stop automatically filtering textures

### ISSUE THREAD

**Issue by skyace65** [Zylann/godot_heightmap_plugin#270]
**Is your feature request related to a problem? Please describe.**
I'm trying to create terrain with pixel art textures. The textures have had filtering disabled in importing options, but the terrain seems to be filtering them.

**Describe the solution you'd like**
Stop automatically filtering textures, or at least make it an option. This also seems redundant given the image import option does exist.

**Describe alternatives you've considered**
Creating another terrain shader myself, but I don't know how to do that.

**Additional context**
None

**Comment by Calinou** [comment]
From a quick search in the codebase, there seems to be options for filtering. They may not work as expected though.

**Comment by Zylann** [comment]
I don't think any of the shaders do filtering at all. Filtering entirely depends on textures you are importing.

**Comment by skyace65** [comment]
I did some more testing and you're right, the terrain isn't filtering the textures.

The issue is terrain isn't recognizing import changes unless you close and open Godot. Even creating a new scene and terrain after reimporting is using the filtered version of the texture.

Any idea on if this is an issue with the plugin or how Godot handles reimporting?

**Comment by skyace65** [comment]
@Zylann I did some more testing and you're right, the terrain isn't filtering the textures.

The issue is terrain isn't recognizing import changes unless you close and open Godot. Even creating a new scene and terrain after reimporting is using the filtered version of the texture.

Any idea on if this is an issue with the plugin or how Godot handles reimporting?

**Comment by Calinou** [comment]
> The issue is terrain isn't recognizing import changes unless you close and open Godot. Even creating a new scene and terrain after reimporting is using the filtered version of the texture.
>
> Any idea on if this is an issue with the plugin or how Godot handles reimporting?

This is a known issue, see https://github.com/godotengine/godot/issues/30302#issuecomment-842578025

**Comment by Zylann** [comment]
I'm still not sure what series of step I have to do to reproduce your problem, so I opened one of my test scenes for multisplat which uses a `.packed_texarr`, turned `filter` off in the Import dock, clicked "Reimport", and filtering properly stopped taking effect on the terrain. I opened another test scene which uses classic4, found one texture, unchecked "filter" in the import dock, clicked "Reimport", and it also took effect. So I'm still not sure where the problem is.
